tests/rbac_test.go

```diff
@@ -0,0 +1,207 @@
+		Expect(err).ToNot(HaveOccurred(), "failed to login with %s", tests.UsernameAdminUser)
+
+		ktests.BeforeTestCleanup()
+		tests.CreateUser(tests.UsernameTestUser)
+	})
+
+	AfterEach(func() {
+		By(fmt.Sprintf("Login with the user %s", tests.UsernameAdminUser))
+		_, err := ktests.RunCommand("oc", "login", "-u", tests.UsernameAdminUser, "-p", "123456")
+		Expect(err).ToNot(HaveOccurred(), "failed to login with %s", tests.UsernameAdminUser)
+
+		tests.DeleteUser(tests.UsernameTestUser)
+
+		By(fmt.Sprintf("Delete identity of user %s", tests.UsernameTestUser))
+		_, err = ktests.RunCommand("oc", "delete", "identity", "allow_all_auth:"+tests.UsernameTestUser)
+		Expect(err).ToNot(HaveOccurred(), "failed to delete identity of %s", tests.UsernameTestUser)
+	})
+
+	Describe("RBAC with default permission", func() {
+		It("should allow to access subresource endpoint.", func() {
+			By(fmt.Sprintf("Login with the user %s", tests.UsernameTestUser))
+			_, err := ktests.RunCommand("oc", "login", "-u", tests.UsernameTestUser, "-p", "123456")
+			Expect(err).ToNot(HaveOccurred(), "failed to login with %s", tests.UsernameTestUser)
+
+			By("Create a namespace")
+			_, err = ktests.RunCommand("oc", "new-project", NamespaceTestSystem)
+			Expect(err).ToNot(HaveOccurred(), "failed to create new project %s", NamespaceTestSystem)
+
+			By("Create a VM with yaml.")
+			_, err = ktests.RunCommand("oc", "create", "-f", vmi.Manifest, "-n", NamespaceTestSystem)
+			Expect(err).ToNot(HaveOccurred(), "Should create VM %q", vmi.Manifest)
+
+			By("Check VMI should create.")
+			_, err = ktests.RunCommand("virtctl", "start", vmi.Name, "-n", NamespaceTestSystem)
+			Expect(err).ToNot(HaveOccurred(), "Should start VMI %s", NamespaceTestSystem)
+
+			By("Check the VMI is running.")
+			Eventually(func() string {
+				output, err := ktests.RunCommand("oc", "get", "vmi", vmi.Name, "--template", "{{.status.phase}}", "-n", NamespaceTestSystem)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred(), "failed to get VMI status: %s: %v", output, err)
+				return output
+			}, time.Minute*10, time.Second*1).Should(Equal("Running"), "VMI %q is still not running", vmi.Name)
+
+			By("Connect via console.")
+			vm, err := virtClient.VirtualMachineInstance(NamespaceTestSystem).Get(vmi.Name, &metav1.GetOptions{})
+			expecter, err := ktests.LoggedInCirrosExpecter(vm)
+			Expect(err).ToNot(HaveOccurred(), "failed to connect to console")
+			expecter.Close()
+
+			By("Checking the response from VNC server.")
+			response, err := tests.VNCConnection(NamespaceTestSystem, vmi.Name)
+			Expect(response).To(Equal("RFB 003.008"))
+			Expect(err).ToNot(HaveOccurred(), "VMI %q should get response from VNC", vmi.Name)
+
+			By("Check VMI should stop.")
+			_, err = ktests.RunCommand("virtctl", "stop", vmi.Name, "-n", NamespaceTestSystem)
+			Expect(err).ToNot(HaveOccurred(), "VMI %q should be scheduled to stop: %v", vmi.Name, err)
+
+			By("Delete a VM with yaml.")
```

Comment:
By the way, is there any specific reason why you use `oc delete -f ${vm_manifest} -n ${namespace}` instead of `oc delete vm ${vm_name} -n ${namespace}`?

```suggestion
			By("Deleting a VM specified by manifest")
```

Reply:
@gszasz There is no specific reason for that. I personally use `oc delete -f ${vm_manifest} -n ${namespace}`